design/fast_finality.md

```diff
@@ -0,0 +1,66 @@
+# Block explorer design
+
+## Scope
+
+The introduction of a centralised sequencer to improve cost-per-transaction during the bootstrapping phase, as 
+described in the [Bootstrapping Strategy design doc](./Bootstrapping_strategy.md).
+
+## Assumptions
+
+* There is a single aggregator that is also the genesis node. This node assumes the role of sequencer
+
+## Constraints
+
+* Rollups are only published to the L1 every `x` blocks, where `x` >> 1
+
+## Design
+
+### Node start-up
+
+At start-up, the host checks if one of the following applies:
+
+* They are the genesis node and are an aggregator
+* They are *not* the genesis node and are *not* an aggregator
+
+If neither of these conditions is met, the host shuts down.
+
+The identity of the sequencer is listed in the management contract on the L1. This allows other nodes to verify that 
+the rollups are created by the sequencer.
+
+### Creation of light batches
+
+On each block, a *light batch* is created. This light batch is formally identical to the rollup of the final design, 
+including a list of finalised transactions and a header including information on the current light batch and the 
+hash of the "parent" light batch. The light batch is signed and encrypted by the sequencer.
+
+As in the final design, each light batch is linked to a specific L1 block, and (at most) a single light batch is 
+created per L1 block. The light batch is immediately distributed by the sequencer to all other nodes. Unlike in the 
+final design, these light batches are not sent to be included on the L1.
+
+From the user's perspective, the transactions in the light batch are considered final (e.g. responses to RPC calls from 
+the client behave as if the transactions were completely final).
+
+### Creation of rollups
+
+Every `x` blocks, the sequencer creates a rollup. This rollup contains all the light batches created since the last 
+rollup, in a Merkle tree structure. This rollup is sent to be included on the L1.  The rollup is signed and encrypted 
```

Comment:
we need to elaborate on how the non-inclusion proof will work.
Maybe not implement it in the first phase, but have a plan for it.
Perhaps we can use a sparse merkle tree for that.

Ideally, the sequencer would be slashed automatically by the management contract if it failed to include a LB that it previously signed.